docs/extensions/substrait.md

```diff
@@ -4,4 +4,109 @@ title: Substrait
+     ```
+3) Blob Consumption
+     
+     To consume a substrait blob the ```from_substrait(blob)``` function must be called with a valid substrait BLOB plan.
+     ```sql
+     CALL from_substrait('\x12\x07\x1A\x05\x1A\x03lte\x12\x11\x1A\x0F\x10\x01\x1A\x0Bis_not_null\x12\x09\x1A\x07\x10\x02\x1A\x03and\x12\x10\x1A\x0E\x10\x03\x1A\x0Acount_star\x1A\xA4\x01\x12\xA1\x01\x0A\x94\x01:\x91\x01\x12\x86\x01"\x83\x01\x12y:w\x12c\x12a\x12+\x0A)\x12\x1B\x0A\x08exercise\x0A\x0Fdificulty_level:\x0A\x0A\x08crossfit\x1A2\x1A0\x08\x02"\x18\x1A\x16\x1A\x14"\x0A\x1A\x08\x12\x06\x0A\x04\x12\x02\x08\x01"\x06\x1A\x04\x0A\x02(\x05"\x12\x1A\x10\x1A\x0E\x08\x01"\x0A\x1A\x08\x12\x06\x0A\x04\x12\x02\x08\x01\x1A\x08\x12\x06\x0A\x04\x12\x02\x08\x01\x1A\x06\x12\x04\x0A\x02\x12\x00\x1A\x00"\x04\x0A\x02\x08\x03\x1A\x06\x12\x04\x0A\x02\x12\x00\x12\x08exercise'::BLOB)
+     ----
+     2
+   ```
+
+### Python
+Before using the extension you must remember to properly load it. To load an extension in python, you must execute the sql commands within a connection.
+```python
+import duckdb
+
+con = duckdb.connect()
+con.execute("INSTALL('substrait')");
+con.execute("LOAD('substrait')")
+```
+
+1) Blob Generation
+     
+     To generate a substrait blob the ```get_substrait(SQL)``` function must be called, from a connection, with a valid SQL select query.
+     ```python
+     con.execute('CREATE TABLE crossfit (exercise text,dificulty_level int);')
+     con.execute("INSERT INTO crossfit VALUES ('Push Ups', 3), ('Pull Ups', 5) , (' Push Jerk', 7), ('Bar Muscle Up', 10);")
+     
+     proto_bytes =  con.get_substrait("select count(exercise) as exercise from crossfit where dificulty_level <=5")     
+   ```
+2) Json Generation
+     
+     To generate a json representing  the substrait plan the ```get_substrait_json(SQL)``` function, from a connection, must be called with a valid SQL select query.
+     ```python
+     json =  con.get_substrait_json("select count(exercise) as exercise from crossfit where dificulty_level <=5").fetchone()[0]
+     ```
+3) Blob Consumption
+     
+     To consume a substrait blob the ```from_substrait(blob)``` function must be called, from the connection, with a valid substrait BLOB plan.
+     ```python
+     query_result = con.from_substrait(proto_bytes)
+    ```
+
+### R
+Before using the extension you must remember to properly load it. To load an extension in R, you must execute the sql commands within a connection.
+```r
+con <- dbConnect(duckdb::duckdb(config=list("allow_unsigned_extensions"="true")))

```

Comment:
This was prolly my mistake before, but since the repo is private this line could be
`con <- dbConnect(duckdb::duckdb())`

and the order should be reversed

```
dbExecute(con, "INSTALL('substrait')"))
dbExecute(con, "LOAD('substrait')")
```